hyperspy/model.py

```diff
@@ -1087,27 +532,27 @@ def _errfunc4mpfit(self, p, fjac=None, x=None, y=None,
             return [0, self._jacobian(p, y).T]
 
     def _calculate_chisq(self):
-        if self.spectrum.metadata.has_item('Signal.Noise_properties.variance'):
+        if self.signal.metadata.has_item('Signal.Noise_properties.variance'):
 
-            variance = self.spectrum.metadata.Signal.Noise_properties.variance
+            variance = self.signal.metadata.Signal.Noise_properties.variance
             if isinstance(variance, Signal):
                 variance = variance.data.__getitem__(
                     self.axes_manager._getitem_tuple)[self.channel_switches]
         else:
             variance = 1.0
-        d = self(onlyactive=True) - self.spectrum()[self.channel_switches]
+        d = self(onlyactive=True) - self.signal()[self.channel_switches]
```

Comment:
Actually - I'm not sure that's correct... Implementing what you suggest breaks the fitting test that I wrote as in the screenshot below. Is this indicative of a more fundamental error in something I've done, or is it just a slip?
![screenshot from 2015-12-12 11-32-19](https://cloud.githubusercontent.com/assets/10629656/11761701/3ff9166e-a0c4-11e5-8625-2ce0eb7bea62.png)